src/Microsoft.ML.Data/Utilities/ColumnCursor.cs

```diff
@@ -20,16 +20,21 @@ public static class ColumnCursorExtensions
         /// </summary>
         /// <typeparam name="T">The type of the values. This must match the actual column type.</typeparam>
         /// <param name="data">The data view to get the column from.</param>
-        /// <param name="env">The current host environment.</param>
-        /// <param name="columnName">The name of the column to extract.</param>
-        public static IEnumerable<T> GetColumn<T>(this IDataView data, IHostEnvironment env, string columnName)
+        /// <param name="column">The column to be extracted.</param>
+        public static IEnumerable<T> GetColumn<T>(this IDataView data, DataViewSchema.Column column)
         {
-            Contracts.CheckValue(env, nameof(env));
-            env.CheckValue(data, nameof(data));
-            env.CheckNonEmpty(columnName, nameof(columnName));
+            Contracts.CheckValue(data, nameof(data));
+            Contracts.CheckNonEmpty(column.Name, nameof(column));
 
-            if (!data.Schema.TryGetColumnIndex(columnName, out int col))
-                throw env.ExceptSchemaMismatch(nameof(columnName), "input", columnName);
+            var colIndex = column.Index;
+            var colType = column.Type;
+            var colName = column.Name;
+
+            // Use column index as the principle address of the specified input column and check if that address in data contains
```

Comment:
I believe this logic is to make sure the `column`  (type DataViewSchema.Column)  is indeed from the same `data`  (type IDataView). Can we add a test that exercises this ?  Perhaps by passing a `column` from a different dataview, and making sure it fails.

I see some tests which exercise the type (given the *same* dataview), but none which exercises the new logic added above

Reply:
Added in Iteration 6.

---
In reply to: [260851354](https://github.com/dotnet/machinelearning/pull/2738#discussion_r260851354) [](ancestors = 260851354)